server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -104,7 +105,7 @@ public AggregationProcessor aggregationProcessor(SearchContext searchContext) {
 
     private static <T extends Exception> void rethrowCauseIfPossible(RuntimeException re, SearchContext searchContext) throws T {
         // Rethrow exception if cause is null
-        if (re.getCause() == null) {
+        if (re.getCause() == null || re instanceof OpenSearchException) {
```

Comment:
We surely need a test for it, @neetikasinghal could you please craft one (from the issue)? thank you.

Reply:
added the test case for this